kuma/users/signup.py

```diff
@@ -34,7 +34,8 @@ class SignupForm(BaseSignupForm):
     terms = forms.BooleanField(label=_(u'I agree'),
                                required=True,
                                error_messages={'required': TERMS_REQUIRED})
-
+    is_github_url_public = forms.BooleanField(label=_(u'I would like to make my github profile URL public'),
+                                              required=False)
```

Comment:
You'll need some more code to save this additional value on the user profile.

``kuma.users.signup.SignupForm`` is derived from django-allauth's [allauth.socialaccount.forms.SignupForm](https://github.com/pennersr/django-allauth/blob/master/allauth/socialaccount/forms.py#L12-L36). This calls:

    user = adapter.save_user(request, self.sociallogin, form=self)

You can see the base version in [allauth.socialaccount.adapter.DefaultSocialAccountAdapter](https://github.com/pennersr/django-allauth/blob/0367b51514e592db011511e420570789c7d1df75/allauth/socialaccount/adapter.py#L70-L82). We override it in [kuma.users.adapters.KumaSocialAccountAdapter](https://github.com/mozilla/kuma/blob/master/kuma/users/adapters.py#L179-L192), but it basically just calls the default version.

That in turn calls [allauth.account.adapter.DefaultAccountAdapter.save_user](https://github.com/pennersr/django-allauth/blob/0367b51514e592db011511e420570789c7d1df75/allauth/account/adapter.py#L221-L248) to actually save the user. We also override this in [kuma.users.adapaters.KumaAccountAdapter](https://github.com/mozilla/kuma/blob/master/kuma/users/adapters.py#L24), but don't customize the ``save_user`` method.

You'll need to add a ``save_user`` method to ``KumaAccountAdapter`` that:

1. Calls the super class's ``save_user`` method with ``commit=False``
2. Adds the ``is_github_url_public`` setting from ``form.cleaned_data``
3. Saves the user
4. Returns the user.

Plus tests.

Reply:
I have a feeling that tests shouldn't write into the database.

Do I need to mock the save_user() to avoid writing into the Db or do we have a specific db for test which is reset everytime ?

I must admit that I don't really know how to start writing this test.

Reply:
If you get the code working, I'll add tests in a new commit. Or, I can add failing tests and then you can write the code to make them work 😄 

Tests run against a nearly empty copy of the database, and are wiped out between tests. Writing to the database slows down the tests, but it is safe, and we'd need more serious refactoring to move away from database writes in tests.

Reply:
I finally managed to get some time to make the code work properly.

I'm sorry it took so long, I'm still trying to understand how I should do the tests.

I guess the needed tests are : 
- [ ] A test for the registration process in itself for the `save_user()` [kuma.users.adapaters.KumaAccountAdapter.save_user](https://github.com/mozilla/kuma/blob/master/kuma/users/adapters.py#L24) in [KumaSocialAccountAdapterTestCase](https://github.com/mozilla/kuma/blob/master/kuma/users/tests/test_adapters.py#L15)
- [ ] A functional test for the registration process in https://github.com/mozilla/kuma/tree/master/tests/functional 
- [ ] A functional test for the profile display in [test_profiles](https://github.com/mozilla/kuma/blob/master/tests/functional/test_profiles.py)